langchain4j-vertex-ai-gemini/src/main/java/dev/langchain4j/model/vertexai/VertexAiGeminiChatModel.java

```diff
@@ -0,0 +1,82 @@
+package dev.langchain4j.model.vertexai;
+
+import com.google.cloud.vertexai.VertexAI;
+import com.google.cloud.vertexai.api.Content;
+import com.google.cloud.vertexai.api.GenerateContentResponse;
+import com.google.cloud.vertexai.api.GenerationConfig;
+import com.google.cloud.vertexai.generativeai.preview.GenerativeModel;
+import com.google.cloud.vertexai.generativeai.preview.ResponseHandler;
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.output.Response;
+import lombok.Builder;
+
+import java.io.IOException;
+import java.util.List;
+
+import static dev.langchain4j.internal.RetryUtils.withRetry;
+import static dev.langchain4j.internal.Utils.getOrDefault;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+
+/**
+ * Represents a Google Vertex AI Gemini language model with a chat completion interface, such as gemini-pro.
+ * See details <a href="https://cloud.google.com/vertex-ai/docs/generative-ai/model-reference/gemini">here</a>.
+ */
+public class VertexAiGeminiChatModel implements ChatLanguageModel {
+
+    private final GenerationConfig generationConfig;
+    private final GenerativeModel model;
+    private final Integer maxRetries;
+
+    @Builder
+    public VertexAiGeminiChatModel(String project,
+                                   String location,
+                                   String modelName,
+                                   Float temperature,
+                                   Integer maxOutputTokens,
+                                   Integer topK,
+                                   Float topP,
+                                   Integer maxRetries) {
+
+        GenerationConfig.Builder generationConfigBuilder = GenerationConfig.newBuilder();
+        if (temperature != null) {
+            generationConfigBuilder.setTemperature(temperature);
+        }
+        if (maxOutputTokens != null) {
+            generationConfigBuilder.setMaxOutputTokens(maxOutputTokens);
+        }
+        if (topK != null) {
+            generationConfigBuilder.setTopK(topK);
+        }
+        if (topP != null) {
+            generationConfigBuilder.setTopP(topP);
+        }
+        this.generationConfig = generationConfigBuilder.build();
+
+        try (VertexAI vertexAI = new VertexAI(
+                ensureNotBlank(project, "project"),
+                ensureNotBlank(location, "location"))
+        ) {
+            this.model = new GenerativeModel(ensureNotBlank(modelName, "modelName"), vertexAI);
```

Comment:
Another thing we can configure (later on, in a future version) is the ability to filter/moderate content. There's the notion of `SafetySettings` in Gemini, which looks like so:

```java
      GenerativeModel model = new GenerativeModel(modelName, generationConfig, vertexAI);

      List<SafetySetting> safetySettings = Arrays.asList(
          SafetySetting.newBuilder()
              .setCategory(HarmCategory.HARM_CATEGORY_HATE_SPEECH)
              .setThreshold(SafetySetting.HarmBlockThreshold.BLOCK_LOW_AND_ABOVE)
              .build(),
          SafetySetting.newBuilder()
              .setCategory(HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT)
              .setThreshold(SafetySetting.HarmBlockThreshold.BLOCK_MEDIUM_AND_ABOVE)
              .build()
      );
```

It's a bit verbose though, but I'm wondering if there's something more generic that could be used across language models?
I think the OpenAI model has a simple moderation flag, for example? Here it's more fine-grained, so not sure how to make something really generic.

Reply:
OpenAI has a separate [moderation endpoint](https://platform.openai.com/docs/guides/moderation/quickstart), did not see an option to set moderation policy right in the completion endpoint. But it is a good idea, I will add an option to ctor to configure this